app/src/features/Joins/JoinSelect.tsx

```diff
@@ -3,51 +3,50 @@ import React from "react";
 import { IconButton, Grid, Typography } from "@material-ui/core";
 import CloseIcon from "@material-ui/icons/Close";
 
-import ColumnSelects from "features/Columns/ColumnSelect";
-import { PendingJoin } from "features/Mappings/mappingSlice";
+import ColumnSelect from "features/Columns/ColumnSelect";
 import { Column } from "services/api/generated/api.generated";
 
 type JoinSelectsProps = {
-  join: PendingJoin;
-  disableDelete?: boolean;
-  onChange?: (join: PendingJoin) => void;
+  columns: Partial<Column>[];
```

Comment:
We can be more specific for `columns` type

 ```suggestion
  columns: [ Partial<Column>, Partial<Column> ]```